### ISSUE TITLE
RLS: 1.0.0

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#27492]
discussion for 1.0. 

We are focusing all deprecation removals and we should try to release by September 2019.

**Comment by jbrockmendel** [comment]
duplicate of #17287

**Comment by jreback** [comment]
not a duplicate

the other issue is a wish list

**Comment by simonjayhawkins** [comment]
> We are focusing all deprecation removals

does this start now, or after 0.25.1, 0.25.2?

**Comment by jreback** [comment]
yes removals start now for 1.0

**Comment by simonjayhawkins** [comment]
also, are we dropping Python 3.5 for 1.0? and moving to inline variable annotations?

**Comment by jreback** [comment]
yes to both (need to drop 3.5 first of course)

**Comment by simonjayhawkins** [comment]
cool.

3.8.0 final: Monday, 2019-10-21, so after pandas 1.0, shouldn't need to backport 3.8 issues to 0.25, but will probably need to backport 3.8 issues to 1.0.

at some point before the 1.0 release, we'll probably need PY3.8 builds against either the betas or RCs

**Comment by WillAyd** [comment]
We don't have a list of removals we need to make here yet right? I assume it's most everything in #18262 and maybe more. Happy to open a new issue for that if not done already and track therein (or here as well)

**Comment by jorisvandenbossche** [comment]
> I assume it's most everything in #18262 

That issue is for things we want to deprecate, not necessarily for things to remove. Up to now, we have been using this issue to keep track of that: https://github.com/pandas-dev/pandas/issues/6581/ 
In principle everything in that list should be removed (maybe with priority for the older ones)

**Comment by TomAugspurger** [comment]
cc @pandas-dev/pandas-core it's about time to cut a release candidate. I'd like to do one on Friday / this weekend if possible, and let it sit for a week or two. Does that sound reasonable to people? I'll go through the backlog of issues / PRs tagged for 1.0, but let me know if you have an issue you consider a blocker and I'll add it to the list.

**Comment by jreback** [comment]
> cc @pandas-dev/pandas-core it's about time to cut a release candidate. I'd like to do one on Friday / this weekend if possible, and let it sit for a week or two. Does that sound reasonable to people? I'll go through the backlog of issues / PRs tagged for 1.0, but let me know if you have an issue you consider a blocker and I'll add it to the list.

sgtm. i don't think we have anything actually blocking, just a couple of PRs (NA stuff) that should merge before, but seems doable.

**Comment by jorisvandenbossche** [comment]
Would targetting the end of next week instead of this week also be OK? (or a few days earlier, if ready) 

I would personally prefer that. There have been merged quite some substantial changes the last days / still need to be merged today/tomorrow, and I think it would be good to have those at least a few days in master (and in a week that more people are active. I am myself still mostly offline until the end of this week).

**Comment by TomAugspurger** [comment]
That’s reasonable. I won’t have much time today either.

**Comment by TomAugspurger** [comment]
We also have our dev meeting on Wednesday. I think we may want to discuss branching things and what we’ll consider merging to master during the RC period. So let’s plan for Wednesday afternoon.

**Comment by TomAugspurger** [comment]
@pandas-dev/pandas-core If there aren't any objections, I'd like to tag the RC at around 2:00 Central (~6 hours from now). What PRs / issues should be considered blockers (feel free to add to the list)?

- [ ] https://github.com/pandas-dev/pandas/pull/30588
- [ ]

**Comment by TomAugspurger** [comment]
OK, tagging the RC.

**Comment by TomAugspurger** [comment]
Tagged, and have started a 1.0.x maintenance branch. Merges to master with the 1.0.0 milestone will be backported to that branch.

Binaries are building.

**Comment by TomAugspurger** [comment]
@cgohlke Not sure if you typically build wheels for RCs, but if you do we've tagged 1.0.0rc0: https://github.com/pandas-dev/pandas/releases/download/v1.0.0rc0/pandas-1.0.0rc0.tar.gz

**Comment by topper-123** [comment]
I know it's not super-critical, but could we backport type hint related PR's to 1.0? I'm thinking especially about #30761 and #30947, both of which are currently tagged for 1.1. There could of course be other.

The rationale is especially:
* Type hints give a better user experience for users with IDEs (PyCharm etc.)
* a very low (or no) risk of breakage, because it's just type hint, not code.

**Comment by TomAugspurger** [comment]
I don't have strong thoughts on backporting the typing PRs. My initial preference is to not backport them, but it's not a strong feeling.

---

@pandas-dev/pandas-core what do people think about targeting 1.0.0 for early next week, perhaps Tuesday the 28th? Add / modify this list as you see fit. 

https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0

**Blockers / Regressions**

- https://github.com/pandas-dev/pandas/issues/31157
- https://github.com/pandas-dev/pandas/issues/30892
- https://github.com/pandas-dev/pandas/issues/30944 (is this a blocker?)
- https://github.com/pandas-dev/pandas/issues/30742 (perf)
- https://github.com/pandas-dev/pandas/issues/31109

**Nice to have**

- https://github.com/pandas-dev/pandas/issues/29884
- https://github.com/pandas-dev/pandas/issues/30889
- https://github.com/pandas-dev/pandas/issues/3942

**Unsure** (move these to where you think they should be)

- https://github.com/pandas-dev/pandas/issues/27357

**Comment by TomAugspurger** [comment]
@pandas-dev/pandas-core I'm hoping to do the release today. The only potential blocker is https://github.com/pandas-dev/pandas/pull/31359, but that's a bit tricky. It's not clear what the best way forward is there (chime in there if you have thoughts).

Is there any objection to also pushing the new website to `pandas.pydata.org` at the same time?

**Comment by TomAugspurger** [comment]
OK, tagging it!

**Comment by TomAugspurger** [comment]
Conda packages are up in defaults and conda-forge. Just waiting on windows wheels, then I'll send out the announcement.

**Comment by TomAugspurger** [comment]
Binaries are up, and sent out the announcement.

Congrats everyone!

**Comment by jorisvandenbossche** [comment]
Thanks a lot, Tom !

**Comment by jreback** [comment]
thanks everyone!
